cmd/process-agent/main_common.go

```diff
@@ -101,44 +95,100 @@ func getSettingsClient(_ *cobra.Command, _ []string) (settings.Client, error) {
 }
 
 func init() {
-	rootCmd.AddCommand(configCommand, app.StatusCmd(), app.VersionCmd)
+	rootCmd.AddCommand(configCommand, app.StatusCmd, app.VersionCmd, app.CheckCmd)
 }
 
 func deprecatedFlagWarning(deprecated, replaceWith string) string {
 	return fmt.Sprintf("WARNING: `%s` argument is deprecated and will be removed in a future version. Please use `%s` instead.\n", deprecated, replaceWith)
 }
 
-// fixDeprecatedFlags modifies os.Args so that non-posix flags are converted to posix flags
-// it also displays a warning when a non-posix flag is found
-func fixDeprecatedFlags(args []string, w io.Writer) {
-	deprecatedFlags := []string{
-		// Global flags
-		"-config", "--config", "-sysprobe-config", "-pid", "-info", "-version", "-check",
-		// Windows flags
-		"-install-service", "-uninstall-service", "-start-service", "-stop-service", "-foreground",
+type replaceFlag struct {
+	hint string
+	args []string
+}
+
+type replaceFlagFunc func(arg string, flag string) replaceFlag
+
+func replaceFlagPosix(arg string, flag string) replaceFlag {
+	newFlag := "-" + flag
+	return replaceFlag{
+		hint: newFlag,
+		args: []string{
+			strings.Replace(arg, flag, newFlag, 1),
+		},
 	}
+}
+
+func replaceFlagExact(replaceWith string) replaceFlagFunc {
+	return func(arg string, flag string) replaceFlag {
+		return replaceFlag{
+			hint: replaceWith,
+			args: []string{strings.Replace(arg, flag, replaceWith, 1)},
+		}
+	}
+}
 
-	replaceFlags := map[string]string{
-		"-config":  "--cfgpath",
-		"--config": "--cfgpath",
+func replaceFlagSubCommandPosArg(replaceWith string) replaceFlagFunc {
+	return func(arg string, _ string) replaceFlag {
+		parts := strings.SplitN(arg, "=", 2)
+		parts[0] = replaceWith
+		return replaceFlag{
+			hint: replaceWith,
+			args: parts,
+		}
+	}
+}
+
+// fixDeprecatedFlags transforms args so that they are conforming to the new CLI structure:
+// - replace non-posix flags posix flags
+// - replace deprecated flags with their command equivalents
+// - display warnings when deprecated flags are encountered
+func fixDeprecatedFlags(args []string, w io.Writer) []string {
+	var (
+		replaceCfgPath = replaceFlagExact("--cfgpath")
+		replaceVersion = replaceFlagExact("version")
+		replaceCheck   = replaceFlagSubCommandPosArg("check")
+	)
+
+	deprecatedFlags := map[string]replaceFlagFunc{
+		// Global flags
+		"-config":          replaceCfgPath,
+		"--config":         replaceCfgPath,
+		"-sysprobe-config": replaceFlagPosix,
+		"-version":         replaceVersion,
+		"--version":        replaceVersion,
+		"-check":           replaceCheck,
+		"--check":          replaceCheck,
+		"-pid":             replaceFlagPosix,
+		"-info":            replaceFlagPosix,
+		// Windows flags
+		"-install-service":   replaceFlagPosix,
+		"-uninstall-service": replaceFlagPosix,
+		"-start-service":     replaceFlagPosix,
+		"-stop-service":      replaceFlagPosix,
+		"-foreground":        replaceFlagPosix,
 	}
 
+	var newArgs []string
 	for i, arg := range args {
```

Comment:
nit: You can remove the index var `i` here and use `arg` in the loop as it looks like you are no longer mutating the args slice anymore.